site/_posts/2023-02-03-single-sign-on-laravel-fusionauth.md

```diff
@@ -0,0 +1,347 @@
+
+# Specify a tenant Id or leave this blank for the default tenant
+FUSIONAUTH_TENANT_ID=<ID FOR YOUR TENANT>
+
+# Replace http://localhost:9011 with the address where the FusionAuth application is running
+FUSIONAUTH_BASE_URL=http://localhost:9011
+
+# Replace http://localhost with the address for your PHP application, if necessary
+# We'll create the route for `/auth/callback` later##
+FUSIONAUTH_REDIRECT_URI=http://localhost/auth/callback
+```
+
+### Creating new fields in the User model
+
+Migrations are the code-based version control for your database. Instead of having to share SQL queries with your coworkers so they can create a table or add a column everytime there is a change in the database, you can write code to programatically do this when running a command.
+
+In Laravel, to create a new migration you just need to run the `artisan make:migration` and inform a meaningful name for it. In the next line, we'll create a migration called `add_fusionauth_fields_user_table`:
+
+```shell
+$ ./vendor/bin/sail artisan make:migration add_fusionauth_fields_user_table
+```
+
+Then, go to your `database/migrations` folder, where you'll find a file there with the name you created and prefixed by today's date (e.g `2023_02_03_123000_add_fusionauth_fields_user_table.php`).
+
+Edit that file to add three new columns to the `users` table: `fusionauth_id`, `fusionauth_access_token` and `fusionauth_refresh_token` and allow `NULL` values in the `password` column.
+
+{% include _callout-tip.liquid content="If you cloned our [GitHub repository](https://github.com/FusionAuth/fusionauth-example-laravel-single-sign-on), you can just copy the contents from [laravel/migration.php](https://github.com/FusionAuth/fusionauth-example-laravel-single-sign-on/blob/main/laravel/migration.php)." %}
+
+```php
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+return new class extends Migration
+{
+    /**
+     * When running this migration, we want to create three new columns and change the password so it can be NULL
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('users',  function (Blueprint $table) {
+            $table->string('fusionauth_id', 36)->unique();
+            $table->text('fusionauth_access_token');
+            $table->text('fusionauth_refresh_token')->nullable();
+            $table->string('password')->nullable()->change();
+        });
+    }
+
+    /**
+     * When rolling back this migration, we want to drop the added columns and revert the password to NOT NULL again
+     *
+     * @return void
+     */
+    public function down()
+    {
+        Schema::table('users',  function (Blueprint $table) {
+            $table->dropColumn('fusionauth_id');
+            $table->dropColumn('fusionauth_access_token');
+            $table->dropColumn('fusionauth_refresh_token');
```

Comment:
Does socialite automatically refresh tokens or do we have to do that ourselves?

Reply:
We have to do that ourselves. I think we can improve [the FusionAuth provider](https://github.com/SocialiteProviders/FusionAuth) to handle this. What do you think?

Reply:
Hmmm. let's let that be for now. Add a text note that "your laravel application will have to refresh tokens when the access token expires", please.

Reply:
Done